rfcs/min-relations.md

```diff
@@ -0,0 +1,273 @@
+    for (mut magic, friends) in query.iter_mut(){
+        // Relations return an iterator when unpacked
+        magic.power += friends.count();
+    }
+}
+
+// Or you can request mutable access to modify the relation's data
+fn interest(query: Query<&mut Relation<Owes>>){
+    for debts in query.iter(){
+        for (owed_to, amount_owed) in debts {
+            println!("we owe {} some money", owed_to);
+            amount_owed *= 1.05; 
+        }
+    }
+}
+
+// You can query for entities that target a specific entity
+fn friend_of_dorothy(
+    mut query: Query<Entity, With<Relation<FriendsWith>>>, 
+    dorothy: Res<Dorothy>,
+) {
+    // Setting relation filters requires that the query is mutable
+    query.set_relation_filters(
+        RelationFilters::new()
+            .add_target_filter::<FriendsWith, _>(dorothy.entity)
+    );
+
+    for source_entity in query.iter(){
+        println!("{} is friends with Dorothy!", source_entity);
+    }
+
+    // RelationFilters are reset at the end of the system, make sure to set them every time your system runs!
+    // You can also reset this manually by applying a new `RelationFilters` to the query
+    // They override existing relation filters, rather than adding to them
+}
+
+// Or even look for some combination of targets!
+fn caught_in_the_middle(mut query: Query<&mut Stress, With<Relation<FriendsWith>>>,
+    dorothy: Res<Dorothy>,
+    elphaba: Res<Elphaba>){
+    
+    // You can directly chain this as set_relation_filters returns &mut Query
+    query.set_relation_filters(
+        // Note that we can set relation filters even if the relation is in a query filter
+        RelationFilters::new()
+            // Note: add_target_filter is additive- i.e. entities this query
+            // matches must be friends with *both* dorothy AND elphaba
+            .add_target_filter::<FriendsWith, _>(dorothy.entity)
+            .add_target_filter::<FriendsWith, _>(elphaba.entity)
+    ).for_each(|mut stress| {
+        println!("{} is friends with Dorothy and Elphaba!", source_entity);
+        stress.val = 1000; 
+    })
+}
+
+/// If we want to combine targets in an "or" fashion, we need a slightly different syntax
+TODO: write example
+
+// Query filters work too!
+fn not_alone(
+    commands: mut Commands, 
+    query: Query<Entity, Without<Relation<FriendsWith>>>,
+    puppy: Res<Puppy>
+) {
+    for lonely_entity in query.iter(){
+        commands.insert_relation(FriendsWith, puppy.entity);
+    }
+}
+
+// So does change detection with Added and Changed!
+fn new_friends(query: Query<&mut Excitement, Added<Relation<FriendsWith>>>){
+    query.for_each(|mut excitement|{excitement.value += 10});
+}
+
+```
+
+TODO: mention relation cleanup on despawn.
+
+Sometimes, we may instead want to look for "entities that are targets of some relation".
+We can do so by iterating over all entities with that relation, and then filtering by target.
+
+```rust
+fn all_children(
```

Comment:
This example is equivalent to `friend_of_dorothy` and could probably be removed.

Reply:
I think I agree. We were originally going to use this to demonstrate reverse relations, but we cut those from min_relations.

Reply:
Removed.